test/unit/test_subprocess.c

```diff
@@ -263,6 +263,120 @@ bool test_interactive(void) {
 	mu_end;
 }
 
+#if HAVE_FORKPTY && HAVE_OPENPTY && HAVE_LOGIN_TTY
+bool test_interactive_pty(void) {
+	rz_subprocess_init();
+	const char *exe_path = get_auxiliary_path("subprocess-interactive");
+
+	RzSubprocessOpt opt = { 0 };
+	opt.file = exe_path;
+	opt.stdin_pipe = RZ_SUBPROCESS_PIPE_NONE;
+	opt.stdout_pipe = RZ_SUBPROCESS_PIPE_NONE;
+	opt.stderr_pipe = RZ_SUBPROCESS_PIPE_NONE;
+	opt.fork_mode = RZ_SUBPROCESS_FORKPTY;
+
+	RzSubprocess *sp = rz_subprocess_start_opt(&opt);
+	mu_assert_notnull(sp, "the subprocess should be created");
+	rz_subprocess_stdin_write(sp, (const ut8 *)"3\n", strlen("3\n"));
+	rz_subprocess_stdin_write(sp, (const ut8 *)"5\n", strlen("5\n"));
+
+	RzStrBuf *sb = rz_subprocess_stdout_readline(sp, UT_TIMEOUT);
+	int c = atoi(rz_strbuf_get(sb));
+	char buf[100];
+	snprintf(buf, sizeof(buf), "%d\n", 3 + 5 + c);
+	rz_subprocess_stdin_write(sp, (const ut8 *)buf, strlen(buf));
+
+	rz_subprocess_wait(sp, UT_TIMEOUT);
+	RzSubprocessOutput *spo = rz_subprocess_drain(sp);
+	mu_assert_streq(remove_cr(spo->out), "Right\n", "A Good message should be returned");
+	mu_assert_eq(spo->ret, 0, "subprocess exited in the right way");
+
+	rz_subprocess_output_free(spo);
+	rz_subprocess_free(sp);
+	rz_subprocess_fini();
+	mu_end;
+}
+
+bool test_interactive_pipe_pty(void) {
+	rz_subprocess_init();
+	const char *exe_path = get_auxiliary_path("subprocess-interactive");
+
+	RzSubprocessOpt opt = { 0 };
+	opt.file = exe_path;
+	opt.stdin_pipe = RZ_SUBPROCESS_PIPE_CREATE;
+	opt.stdout_pipe = RZ_SUBPROCESS_PIPE_CREATE;
+	opt.stderr_pipe = RZ_SUBPROCESS_PIPE_CREATE;
+	opt.fork_mode = RZ_SUBPROCESS_FORKPTY;
```

Comment:
I don't remember anymore, but what is the diff between PIPE_NONE + FORKPTY and PIPE_CREATE + PTY? In other words, what is the diff between this test and test_interactive_pty?

Reply:
But currently, `test_interactive_pipe_pty` passes but doesn't do what I specified above. It directly connects to the slave file descriptor. I'm working on changing this.

Reply:
Upon further thinking, I can't seem to think of a use case where using pipes with forkpty would be any useful. I will just not use pipes if the `fork_mode` is set to `RZ_SUBPROCESS_FORKPTY`. Please let me know if there are any issues with it. Thanks!